src/presentation/controllers/device-controller.ts

```diff
@@ -66,21 +86,18 @@ export class DeviceController extends BaseController {
     }
   }
 
-  private toDeviceDto(device: Device): DeviceDtoInterface | undefined {
-    if (this.isINewsDevice(device)) {
-      return new INewsDeviceDto(device)
-    }
-    if (this.isTelemetricsDevice(device)) {
-      return new TelemetricsDeviceDto(device)
-    }
-    return undefined
-  }
+  @PutRequest('/:deviceId')
+  public async updateDevice(request: Request, response: Response): Promise<void> {
+    try{
+      const device: Device = await request.body
+      const { deviceId } = request.params
 
-  private isINewsDevice(device: Device): device is INewsDevice {
-    return device.type === DeviceType.INEWS
-  }
+      if(device.id !== deviceId) throw new ConflictException('Conflict: The ID in the URL path does not match the ID in the request body.')  //to be in compliance with the REST convention (it is not a strict rule) we pass in the ID, and then we validate it against the body ID to stop IDOR attacks 
```

Comment:
Suggest to move the comment on top of the function, so it's more visible 😄 

```suggestion
      /* To be in compliance with the REST convention (it is not a strict rule) we pass in the ID,
       and then we validate it against the body ID to stop IDOR attacks */
      if (device.id !== deviceId) throw new ConflictException('Conflict: The ID in the URL path does not match the ID in the request body.')
```